src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -129,9 +297,12 @@ class Factory(Subsystem):
     @classmethod
     def register_options(cls, register):
       super(PexBuilderWrapper.Factory, cls).register_options(register)
-      register('--setuptools-version', advanced=True, default='40.6.3',
+      register('--setuptools-version', advanced=True, default='40.6.3', fingerprint=True,
                help='The setuptools version to include in the pex if namespace packages need to be '
                     'injected.')
+      register('--generate-ipex', type=bool, default=False, fingerprint=True,
+               help='Whether to generate a `.ipex` file, which will "hydrate" its dependencies '
+                    'when it is first executed.')
```

Comment:
Would be helpful to expand upon this, e.g. 

> Whether to generate a `.ipex` file, which will "hydrate" its dependencies when it is first executed, rather than at build time (the normal Pex behavior). This option is useful if you want to ____, however, it comes at the cost of ____.

Reply:
Expanded!

Reply:
Thank you so much for that template!!